File: 01-Investigacion/CSharp/bubble_sort.cs
using System.Diagnostics;

class Program {
    static void BubbleSort(int[] array) {
        int len_arr = array.Length; // Longitud del array
        int temp; // Variable temporal para el intercambio
        for (int i = 0; i < len_arr; i++) { // Primer loop
            for (int j = 0; j < len_arr - i - 1; j++) { // Segundo loop
                if (array[j] > array[j + 1]) { // Cambia > por < para orden descendente
                    // Intercambio de elementos
                    temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }

    static void Main() {
        int[] array = new int[10000];
        Random random = new();

        for (int i = 0; i < array.Length; i++) {
            array[i] = random.Next(1, 100001); // Genera números aleatorios entre 1 y 100000
        }

        Stopwatch time = new();
        time.Start();
        BubbleSort(array);
        time.Stop();

        foreach (int item in array) {
            Console.Write(item + " ");
        }

        Console.WriteLine("\nTiempo de ejecución en milisegundos: " + time.ElapsedMilliseconds);
    }
}

File: 01-Investigacion/Python/bubblesort.py
import time
import random

# Bubble sort en Python
def bubbleSort(array):
    len_arr = len(array) # Guardo la longitud del array
    for i in range(len_arr): # Primer loop que recorre todo el array
        for j in range(0, len_arr - i - 1): # Segundo loop comparando los elementos adyacentes entre si
            if array[j] > array[j + 1]: # Cambiar > a < para hacer orden descendente
                arr[j], arr[j+1] = arr[j+1], arr[j] # Hace un cambio entre los elementos del array

if __name__ == "__main__":
    arr = [random.randint(1,100000) for _ in range(10000)]
    start_time = time.time()
    bubbleSort(arr)
    end_time = time.time()
    print(arr)
    print("Tiempo de ejecución en segundos:", end_time - start_time)
